stout/flags.cc

```diff
@@ -113,6 +120,57 @@ void Parser::Parse(int* argc, const char*** argv) {
     values.emplace(name, value);
   }
 
+  // Parse environment variables if environment_variable_prefix_
+  // contains the specific prefix of variables we want to parse.
+  if (environment_variable_prefix_) {
+    // Run through all environment variables and select those which has
+    // prefix (environment_variable_prefix_) in the name. Then we grab
+    // correct name and the value and store this pair in `values`.
+    // We use global variable `environ` to have access to the environment
+    // variables to be able to select all necessary variables for
+    // parsing according to the included prefix (see class `Parser`
+    // in `stout/flags.h` which has `environment_variable_prefix_`
+    // variable of type `std::optional<std::string>`).
+    for (char** env = environ; *env != nullptr; ++env) {
+      if (!absl::StrContains(*env, *environment_variable_prefix_)) {
+        continue;
+      }
+
+      // By default all environment variables are as follows:
+      //    name=value
+      //  Hence we're splitting by '=' to correctly get the name and value.
+      const std::vector<std::string> name_value =
+          absl::StrSplit(*env, absl::MaxSplits('=', 1));
+
+      CHECK_EQ(name_value.size(), 2u)
+          << "Expecting all environment variables to have '=' delimiter";
+
+      // Grab the name and the value.
+      absl::string_view name{name_value[0]};
+      std::string value{name_value[1]};
+
+      if (absl::ConsumePrefix(&name, *environment_variable_prefix_)) {
+        // Check if the prefix of environment variable and the rest part of the
+        // environment variable are separated by underscore '_'.
+        if ((*environment_variable_prefix_).back() != '_') {
+          if (name.at(0) != '_') {
+            std::cerr << "Underscore should separate prefix "
+                      << *environment_variable_prefix_
+                      << " and the rest part of the environment variable name"
+                      << std::endl;
+            exit(1);
+          } else {
+            name.remove_prefix(1);
+          }
+        }
```

Comment:
```suggestion
```

We don't want to get a false positive is someone has another environment variable that has the same prefix name but it not meant as an argument! So we should just ignore them!

So if we have a `PREFIX_FOO=bar` then we'll have a `name = foo` and `value = bar` but if we have a `PREFIXFOO=bar` then we'll just skip it. And if we have a `PREFIX__FOO=bar` then we'll have `name = _foo` and `value = bar`.

Make sense?